hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/Path.java

```diff
@@ -214,6 +213,12 @@ public Path(String pathString) throws IllegalArgumentException {
     // uri path is the rest of the string -- query & fragment not supported
     String path = pathString.substring(start, pathString.length());
 
+    // add "./" in front of Linux relative paths so that a path containing
```

Comment:
Try to avoid code duplication by introducing `initialize(scheme, authority, path)` that both `Path(String)` and `Path(String, String, String)` may use.

Reply:
@yufeldman Also, it is not clear why is this change necessary? The comment is misleading as `scheme` is already initialized.

Reply:
Imagine string passed to that ctor is: `a:b` so with new changes `a:` is not really a scheme.
Now when you try to construct URI with null scheme and authority URI will interpret `a:` as scheme unless path is started with `/`

Reply:
Should there be an explicit check for `null` scheme? Also, how ":/abc" is handled? Should it be considered as "abc" under ":", instead of an empty scheme that is not valid?

Reply:
Please check URI ctor for the behaviors you mentioned. Precautions I am taking is to avoid exceptions in URI ctor.
Also I suggest you look at the doc I attached to JIRA (below is a paragraph from the doc):
There are couple of cases that will not be handled by this approach or behavior will change.
Namely:
1. URI that consists only of the scheme - e.g. “x:” will result in Path: “./x:” (Today trying to
create a URI from this string will result in error: “Relative path in absolute URI”)
2. Path that has colon as last character before “/” - e.g. “abc:/bcd”. This will be interpreted
as scheme “abc” and path “/bcd” - the same as today
3. String such as: “urn:example:animal:ferret:nose” will be interpreted as
“./urn:example:animal:ferret:nose” (Today trying to create a URI from this string will
result in error: “Relative path in absolute URI”)

Reply:
I refer to a check for `scheme == null` prior to making path relative on Linux (`path = "./ + path`).
According to your specification (item 2), ":/abc" will be interpreted as "/abc" with an empty scheme and still throw an exception. Will it be better to handle it as "./:/abc" path?